app/lib/filters/text_filter.rb

```diff
@@ -1,7 +1,27 @@
 module Filters
   class TextFilter < Filter
     def value
-      Array(params)
+      Array(parsed_value)
+    end
+
+  private
+
+    def parsed_value
+      return if params.blank?
+
+      if multi_value?
+        option_lookup.select { |key, _| params.include? key }.values.flatten
+      else
+        params
+      end
```

Comment:
I think this if statement could be written as...

```
return option_lookup.select { |key, _| params.include? key }.values.flatten if multi_value?

params
```

Reply:
I find @karlbaker02 implementation easy to read because if I skim through the code I can see that this method could return two possible things depending on a condition. I don't have read the whole line only to discover that its only likely to return that value if a condition is met.

Reply:
I would normally be with you @andysellick on this kind of defensive programming but I'm not sure what the Ruby convention is on that front - I've never seen it done elsewhere and am of a mind that it perhaps is a little easier to understand with the `if else`

Reply:
Well, if you both think so - I was just trying to reduce the code a little.